alacritty/src/display.rs

```diff
@@ -471,7 +473,18 @@ impl Display {
 
             self.renderer.with_api(&config.ui_config, config.cursor, &size_info, |mut api| {
                 // Iterate over all non-empty cells in the grid.
-                for cell in grid_cells {
+                for mut cell in grid_cells {
+                    // Invert the active match in vi-less search.
+                    if search_state.regex().is_some()
+                        && search_state.focused_match().contains(&Point::new(
+                            display_offset + size_info.screen_lines().0 - cell.line.0 - 1,
```

Comment:
The problem is that we don't have access to grid or terminal here, so we have no way to use their methods for conversion.

The only option would be to convert the `search_state.focused_match()` range to a `Point` by using the `clamp_buffer_to_visible` method of the grid, but that would require deconstructing and reconstructing the range which would be rather ugly too.

Would look something like this, do you prefer that?

```rust
let start = term.grid().clamp_buffer_to_visible(search_state.focused_match().start());
let end = term.grid().clamp_buffer_to_visible(search_stat.focused_match().end());
let viewport_match = start..=end;

// ...

if search_state.regex().is_some() && viewport_search.contains(&Poin::new(cell.line, cell.column)) {
```

Reply:
Yes, I'll prefer it, since you don't reconstruct terminal's internal function implementation, meaning you'll have less chance to break this code if internal function will change.

Reply:
Yeah, makes sense. This way it just duplicates a bunch of glue code, but no logic.